test/jdk/tools/jmod/JmodTest.java

```diff
@@ -920,6 +956,20 @@ static void assertSameContent(Path p1, Path p2) {
         }
     }
 
+    static void assertMagic(Path file, int minorVersion) throws IOException {
```

Comment:
testLibs and testStaticLibs checking the magic number is good, just might be better to have the two call assertMagic with both the expected major and minor version rather than just the minor version.